tests/static-check/run.sh

```diff
@@ -25,7 +25,30 @@ function create_image() {
 
 set -x
 
-# TODO: check how old the image is and recreate if it's too old
+# Get the version of CFEngine from configure
+VERSION=`./configure --version | head -n1 | grep -Eo '[0-9]+\.[0-9]+'`
+VERSION="${VERSION}.x"
+
+# Check if we can find this file upstream
+STATUS=`curl -o /dev/null --silent -Iw '%{http_code}' "https://raw.githubusercontent.com/cfengine/core/refs/heads/${VERSION}/tests/static-check/run.sh"`
+if [[ $STATUS = "404" ]]
+then
+  # If a branch with that version does not exist, then it's probably master
+  VERSION="master"
+  STATUS=`curl -o /dev/null --silent -Iw '%{http_code}' "https://raw.githubusercontent.com/cfengine/core/refs/heads/${VERSION}/tests/static-check/run.sh"`
+fi
+
+# Compare digest of this file with upstream. If it has changed, then we rebuild the container
+if [[ $STATUS = "200" ]]
+then
+  SUM_A=`curl "https://raw.githubusercontent.com/cfengine/core/refs/heads/${VERSION}/tests/static-check/run.sh" | sha256sum | awk '{print $1}'`
+  SUM_B=`sha256sum "$0" | awk '{print $1}'`
```

Comment:
It seems like it might be "easier" to push the sha of the script into the image and then check those against each other.

Checking the sha of this script against upstream seems a bit round-a-bout and not to-the-point.

One case I think would work better with my suggested strategy is if you are iterating on a PR and changing this file, then the first run you would get a new image created. But what if you run this through CI many times after that to make changes to things OTHER than this file? It would still be re-creating the image each time when it doesn't need to.

The scripting looks good just not sure it's quite the right strategy.

Reply:
Yup, that makes sense!